core/common/mp_utils.cpp

```diff
@@ -12,22 +12,31 @@ namespace kagome::common {
   namespace detail {
     template <size_t size, typename uint>
     std::array<uint8_t, size> uint_to_bytes(uint &&i) {
-      auto count = i.backend().size();
-      auto tsize = sizeof(boost::multiprecision::limb_type);
-      auto copy_count = count * tsize;
+      using boost::multiprecision::cpp_int;
+      std::vector<uint8_t> bytes;
+      bytes.reserve(size);
+      export_bits(i, std::back_inserter(bytes), 8);
```

Comment:
pls refactor. consider using additional parameter msg_first = false to get rid of reversion, or even use arithmetical method without using heavy methods.
Anyway try to get rid of redundant vector, writing directly to resulting std::array.